src/Raven.Client/Documents/Operations/GetAttachmentOperation.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Http;
+using Raven.Client.Util;
+using Sparrow.Json;
+
+namespace Raven.Client.Documents.Operations
+{
+    public class GetAttachmentOperation : IOperation<AttachmentResult>
+    {
+        private readonly string _documentId;
+        private readonly string _name;
+        private readonly Action<Stream> _handleStreamResponse;
+
+        public GetAttachmentOperation(string documentId, string name, Action<Stream> handleStreamResponse)
+        {
+            _documentId = documentId;
+            _name = name;
+            _handleStreamResponse = handleStreamResponse;
+        }
+
+        public RavenCommand<AttachmentResult> GetCommand(DocumentConventions conventions, JsonOperationContext context, HttpCache cache)
+        {
+            return new GetAttachmentCommand(_documentId, _name, _handleStreamResponse);
+        }
+
+        private class GetAttachmentCommand : RavenCommand<AttachmentResult>
+        {
+            private readonly string _documentId;
+            private readonly string _name;
+            private readonly Action<Stream> _handleStreamResponse;
+
+            public GetAttachmentCommand(string documentId, string name, Action<Stream> handleStreamResponse)
```

Comment:
I'm not sure that I'm following. We do return here the AttachmentResult. Do you want it to be passed in the constructor? The content type is not needed there.